src/pages/user_profile/AffiliationAndRoles.js

```diff
@@ -115,6 +116,8 @@ export default function AffiliationAndRole(props) {
     };
 
     let updatedUser = await User.updateSelf(payload);
+    const user = await User.getMe();
+    setUserRoleStatuses(user, Storage);
```

Comment:
`updatedUser` should be the same object as `user` in this context so you probably don't have to make the call to `User.getMe()`. Can you verify this and maybe simplify?